src/plugins/explore/public/components/top_nav/top_nav.tsx

```diff
@@ -150,30 +150,33 @@ export const TopNav = ({ setHeaderActionMenu = () => {}, savedExplore }: TopNavP
 
   const dispatch = useDispatch();
 
-  const handleDatasetSelect = (newDataset: any) => {
```

Comment:
@kavilla why is this `any`? I tried `Dataset` but `newDataset.dataSource` will be invalid